patternfly/patternfly-elements
# test: framework testing

## Framework Testing
Running all of our unit tests against React and Vue.

### What has changed and why
I've added Vue and React as devDependencies so we can make sure our components work in Vue and React. Figuring out how to add Angular proved to be too difficult for the MVP. We'll have to research how to add Angular in the future.

The idea for framework testing ended up being pretty simple. Each test directory now includes the normal HTML tests and we've added Vue and React tests.

```
WCT.loadSuites([
  'pfe-markdown_test.html',
  'pfe-markdown_react_test.html',
  'pfe-markdown_vue_test.html'
]);
```
I've also extracted the JavaScript from the original test files so we can reuse the JavaScript tests in the HTML, React, and Vue files.

For each component, the test folder structure should look like this (using pfe-markdown as an example).

```
test
  - index.html
  - pfe-markdown_react_test.html
  - pfe-markdown_test.html
  - pfe-markdown_test.js
  - pfe-markdown_vue_test.html
```

### Testing instructions
1. Pull down the branch
2. `npm install`
3. `npm test`
4. Make sure all the tests pass

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
